ibis/backends/flink/__init__.py

```diff
@@ -331,13 +335,30 @@ def create_table(
             # INTO ... VALUES to keep things consistent
             self._table_env.create_temporary_view(qualified_name, table)
         if isinstance(obj, ir.Table):
-            # TODO(chloeh13q): implement CREATE TABLE for expressions
-            raise NotImplementedError
-        if schema is not None:
-            if not tbl_properties:
-                raise exc.IbisError(
-                    "tbl_properties is required when creating table with schema"
+            if schema is not None and not schema.equals(obj.schema()):
+                raise exc.IbisTypeError(
+                    "Provided schema and Ibis table schema are incompatible. Please "
+                    "align the two schemas, or provide only one of the two arguments."
                 )
+
+            # TODO (mehmet): Is it OK to return a view here?
+            table = ibis.memtable(obj, schema=obj.schema())
+            query_expression = self.compile(table)
+            return self.create_view(
+                name=name,
+                query_expression=query_expression,
+                database=database,
+                catalog=catalog,
+                temporary=temp,
+                overwrite=overwrite,
+            )
+
+        if schema is not None:
+            tbl_properties = tbl_properties or {
+                "connector": "filesystem",
+                "path": "file:///tmp",
+                "format": "parquet",
+            }
```

Comment:
We had discussed potentially making `tbl_properties` optional, right (not necessarily in this PR)? If that's the case, would you still have this block? I don't see other backends referencing `file:///tmp` or something in a hardcoded way.

Reply:
Good question. I have actually spent quite some time looking for a way to create a table without the `connector` option, but could not find any way. In my current understanding, tables have to be based on a "persistent" storage backend. There does not seem to be a "special connector" to data in memory. The closest to this is the [managed table](https://nightlies.apache.org/flink/flink-table-store-docs-release-0.1/docs/development/overview/#managed-table) as far as I could find.

Reply:
> I have actually spent quite some time looking for a way to create a table without the connector option, but could not find any way.

Do you mean in Flink? I think it is technically possible, as you mentioned, if the table is a managed one. I'm on the same page as Deepyaman; I'm also not sure that I understand this implementation / agree that this is a clean implementation here.

Reply:
Thanks. Then we have two options:
1. Revert the change, so we have the following defined in `create_table()`:
```python
        if schema is not None:
            if not tbl_properties:
                raise exc.IbisError(
                    "tbl_properties is required when creating table with schema"
                )
```
which will lead to failures in > 20 tests.

2. Implement the change with [managed table](https://nightlies.apache.org/flink/flink-table-store-docs-release-0.1/docs/development/overview/#managed-table). However, enabling managed table will require additional steps/changes as described [here](https://nightlies.apache.org/flink/flink-table-store-docs-release-0.1/docs/development/overview/#setup-table-store).

@chloeh13q @deepyaman Which option makes more sense to you?

Reply:
I agree with @deepyaman .